src/java/org/apache/cassandra/index/sai/disk/v6/V6OnDiskFormat.java

```diff
@@ -51,4 +51,12 @@ public IndexFeatureSet indexFeatureSet()
     {
         return v6IndexFeatureSet;
     }
+
+    @Override
+    public int jvectorFileFormatVersion()
+    {
+        // Before version EB, we used JVector format 2. Version DC introduced format 4, so we can start using it
```

Comment:
according to my testing here https://github.com/riptano/cndb/issues/14247#issuecomment-2943230825 JVector 4.0.0.beta1 shipped with the March release does not support format 4 and at that time the default version was DC

we can map format 4 only to ED or later

Reply:
This is fixed now, please take a look.